components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -3174,11 +3203,16 @@ public void deleteIdPByResourceId(String resourceId, int tenantId, String tenant
                 throw new IdentityProviderManagementException(String.format(msg, resourceId, tenantDomain));
             }
             deleteIdP(dbConnection, tenantId, null, resourceId);
+            // Delete IdP related secrets from the IDN_SECRET table.
+            IdpMgtServiceComponentHolder.getInstance().getIdPSecretsProcessorService().deleteIdPSecrets(identityProvider);
             IdentityDatabaseUtil.commitTransaction(dbConnection);
         } catch (SQLException e) {
             IdentityDatabaseUtil.rollbackTransaction(dbConnection);
             throw new IdentityProviderManagementException("Error occurred while deleting Identity Provider of tenant "
                     + tenantDomain, e);
+        } catch (SecretManagementException e) {
+            throw new IdentityProviderManagementException("Error while deleting IDP secrets of Identity provider in " +
```

Comment:
Shall we add the IdP name here as well?

Reply:
Resolved with [b272f60](https://github.com/wso2/carbon-identity-framework/pull/4430/commits/b272f6000e56cbc7b8514e76271edf82aec1b059)